### PR TITLE
feat(sample-gen): yaml sample generation

### LINKED ISSUE #6857: Are there any plans to add support for yaml media type examples?
Are there any plans to add support for yaml media type examples? Currently examples given as schema objects will render as either json or xml, but to get them to display as yaml I must copy and paste a separate example as a multi-line string.

Here is an example spec demonstrating the different behaviours:
```yaml
openapi: 3.0.0
info:
  version: '0.1'
  title: Yaml vs Json Examples
  description: ''
paths:
  /resource1:
    post:
      summary: Add some resource - examples are json or xml
      operationId: myOp1
      responses:
        '201':
          description: resource created
        '400':
          description: Invalid input
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/ModelA'
            examples:
              willRenderAsJson:
                $ref: '#/components/examples/example1'
          application/yaml:
            schema:
              $ref: '#/components/schemas/ModelA'
            examples:
              willRenderAsJson:
                $ref: '#/components/examples/example1'
          application/xml:
            schema:
              $ref: '#/components/schemas/ModelA'
            examples:
              willRenderAsXml:
                $ref: '#/components/examples/example1'                 
        description: Resource to add 
  /resource2:
    post:
      summary: Add some resource - examples are string or yaml
      operationId: myOp2
      responses:
        '201':
          description: resource created
        '400':
          description: Invalid input
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/ModelA'
            examples:
              willRenderAsString:
                $ref: '#/components/examples/example2'
          application/yaml:
            schema:
              $ref: '#/components/schemas/ModelA'
            examples:
              willRenderAsYaml:
                $ref: '#/components/examples/example2' 
          application/xml:
            schema:
              $ref: '#/components/schemas/ModelA'
            examples:
              willRenderAsYaml:
                $ref: '#/components/examples/example2'          
        description: Resource to add         
components:
  examples:
    example1:
      summary: Example as schema object
      value:
        PropertyA: foo
        PropertyB: bar
    # to get example1 to show as YAML have to copy paste 
    # and pipe the value
    example2:
      summary: Example as literal YAML string
      value: |
        PropertyA: foo
        PropertyB: bar        
  schemas:
    ModelA:
      type: object
      properties:
        PropertyA:
          type: string
        PropertyB:
          type: string
```
Thanks

_Originally posted by @deltafsdevelopment in https://github.com/swagger-api/swagger-ui/issues/6470#issuecomment-763540384_

### PR DESCRIPTION
**Pull request by mathis-m**
### Description

if content / media type matches yaml or yml it will generate stringified sample like for json
in addition to that it will proccess the json sample vis js-yaml to the final yaml sample result

### Motivation and Context

Fixes #6857
_Originally requested by @deltafsdevelopment in https://github.com/swagger-api/swagger-ui/issues/6470#issuecomment-763540384_
### How Has This Been Tested?

browser
Todo: cypress tests

### Screenshots (if appropriate):

image

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tim-lai** [comment]
So... while `application/yaml` isn't an actual media type, and that there exists multiple ways to express yaml, e.g. 'text/yaml', based on the `xml` implementation as precedence, we can move forward with this as a new feature.

**Review by tim-lai — CHANGES_REQUESTED** [review]
We should add a test for media-type 'text/yaml'

**Comment by mathis-m** [comment]
@tim-lai done

File: test/unit/core/utils.js [hunk 1]
```diff
@@ -1558,6 +1558,22 @@ describe("utils", () => {
...
+
+    it("should return yaml example if yaml is contained in the content-type", () => {
+      const res = getSampleSchema({
+        type: "object",
+      }, "text/yaml", {}, {test: 123})
+
+      expect(res).toEqual("test: 123")
+    })
+
+    it("should return yaml example if yml is contained in the content-type", () => {
```
Comment (tim-lai) [inline test/unit/core/utils.js:1570]: @mathis-m This looks like a duplicate test?

Comment (mathis-m) [reply test/unit/core/utils.js:1570]: @tim-lai just a little difference to ensure `yaml|yml`

**Review by tim-lai — APPROVED** [review]

**Comment by tim-lai** [comment]
@mathis-m Thanks again! This PR is now merged!

**Pull request merged**